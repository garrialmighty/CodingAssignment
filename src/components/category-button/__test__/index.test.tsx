import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';

import CategoryButton from '../';

describe('Component: IconButton', () => {
  it('should render a CategoryButton', () => {
    const {toJSON} = render(<CategoryButton title="Mock Title" />);
    expect(toJSON).toMatchSnapshot();
  });

  it('can handle press events', () => {
    const mockPress = jest.fn();
    const {getByTestId} = render(
      <CategoryButton title="Mock Title" onPress={mockPress} />,
    );
    fireEvent.press(getByTestId('category-button-Mock Title'));
    expect(mockPress).toHaveBeenCalledTimes(1);
  });
});
