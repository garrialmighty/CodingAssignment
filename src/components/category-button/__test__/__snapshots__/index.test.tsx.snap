// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: IconButton should render a CategoryButton 1`] = `[Function]`;
